apps/x509.c

```diff
@@ -793,6 +810,15 @@ int x509_main(int argc, char **argv)
         if (x == NULL)
             goto end;
     }
+
+    out = bio_open_default(outfile, 'w', outformat);
+    if (out == NULL)
+        goto end;
+
+ cert_loop:
+    if (bundle && sk_X509_num(certs) > 0)
```

Comment:
Better move the `sk_X509_num(certs) > 0` check up, doing right after loading the certs:
```
if (sk_X509_num(certs) <= 0)
    goto end;
```

Reply:
Agreed.